### PR TITLE
[Modules] SynapseWorkspaces add workspaceRepositoryConfiguration property

### PR DESCRIPTION
**Pull request by robbertwortell**
# Description

added a parameter to the synapse workspace to add GIT configuration
https://github.com/Azure/ResourceModules/issues/3092

# Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist

- [v] I'm sure there are no other open Pull Requests for the same update/change
- [?] My corresponding pipelines / checks run clean and green without any errors or warnings
- [v] My code follows the style guidelines of this project
- [v] I have commented my code, particularly in hard-to-understand areas
- [v] I have made corresponding changes to the documentation (readme)
- [v] I did format my code

### CODE REVIEW

File: modules/Microsoft.Synapse/workspaces/deploy.bicep [hunk 1]
```diff
@@ -78,6 +78,9 @@ param sqlAdministratorLogin string
 #disable-next-line secure-secrets-in-params // Not a secret
 param sqlAdministratorLoginPassword string = ''
 
+@description('IOptional. An Object with the paramters to add GIT integration')
```
Comment (eriqua) [inline modules/Microsoft.Synapse/workspaces/deploy.bicep:81]:
```suggestion
@description('Optional. Git integration settings.')
```

File: modules/Microsoft.Synapse/workspaces/deploy.bicep
Comment (eriqua) [inline modules/Microsoft.Synapse/workspaces/deploy.bicep:1]: Please regenerate the readme file leveraging the [Set-ModuleReadme](https://github.com/Azure/ResourceModules/blob/main/utilities/tools/Set-ModuleReadMe.ps1) utility

Comment (robbertwortell) [reply modules/Microsoft.Synapse/workspaces/deploy.bicep:1]: Do you have an example how to run it?

Comment (eriqua) [reply modules/Microsoft.Synapse/workspaces/deploy.bicep:1]: Hey @robbertwortell, sure: from the repo root folder `ResourceModules` do the following in PS:

- Import the utility with 
```powershell
. ./utilities/tools/Set-ModuleReadMe.ps1
``` 

- Run the function to update the readme file using the path to the module you updated
```powershell
$templatePaths = (Get-ChildItem 'modules\Microsoft.Synapse\workspaces' -Filter 'deploy.bicep' -Recurse).FullName
$templatePaths | ForEach-Object { Set-ModuleReadMe -TemplateFilePath $_ }
``` 

- Commit and push changes

Comment (eriqua) [reply modules/Microsoft.Synapse/workspaces/deploy.bicep:1]: Hey @robbertwortell, could you double check which Bicep version you have installed for both az cli and PS?
To check, please run:
```powershell
az bicep version
bicep --version
``` 
They should be both at min 0.15.31.
To upgrade, you can follow guidelines [here](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/install#azure-cli) and [here](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/install#azure-powershell)

**Comment by robbertwortell** [comment]
@microsoft-github-policy-service agree
Met vriendelijke groet / Kind regards,​​​​
Robbert Bonefaas
.
Bekijk
ook eens:​
***@***.***<https://www.wortell.nl/nl/themas/smart>
Consultant
.
+31651455720<tel:+31651455720>

**Review by robbertwortell — COMMENTED** [review]
docs are also created

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**